universal-login-relayer/test/relayer/services/IdentityService.js

```diff
@@ -1,156 +1,113 @@
-    hooks = new EventEmitter();
-    authorisationService = new AuthorisationService();
-    identityService = new IdentityService(wallet, ensService, authorisationService, hooks, provider);
+  beforeEach(async () => {
+    ({wallet, provider, identityService, callback, mockToken, authorisationService, otherWallet} = await testHelper.load(basicIdentityService));
+    transaction = await identityService.create(wallet.address, 'alex.mylogin.eth');
+    contract = await waitForContractDeploy(wallet, Identity, transaction.hash);
+    msg = {...transferMessage, from: contract.address, gasToken: mockToken.address};
+    await wallet.send(contract.address, utils.parseEther('1.0'));
+    await mockToken.transfer(contract.address, utils.parseEther('1.0'));
   });
 
-  describe('IdentityService', async () => {
-    let contract;
-    let callback;
-    let transaction;
+  describe('Create', async () => {
+    it('returns contract address', async () => {
+      expect(contract.address).to.be.properAddress;
+    });
 
-    before(async () => {
-      callback = sinon.spy();
-      hooks.addListener('created', callback);
-      transaction = await identityService.create(managementKey.address, 'alex.mylogin.eth');
-      contract = await waitForContractDeploy(managementKey, Identity, transaction.hash);
-      await contract.setRequiredApprovals(0);
+    it('is initialized with management key', async () => {
+      const managementKeys = await contract.getKeysByPurpose(MANAGEMENT_KEY);
+      const expectedKey = wallet.address.slice(2).toLowerCase();
+      expect(managementKeys).to.have.lengthOf(1);
+      expect(managementKeys[0]).to.endsWith(expectedKey);
     });
 
-    describe('Create', async () => {
-      it('returns contract address', async () => {
-        expect(contract.address).to.be.properAddress;
-      });
+    it('has ENS name reserved', async () => {
+      expect(await provider.resolveName('alex.mylogin.eth')).to.eq(contract.address);
+    });
 
-      it('is initialized with management key', async () => {
-        const managementKeys = await contract.getKeysByPurpose(MANAGEMENT_KEY);
-        const expectedKey = managementKey.address.slice(2).toLowerCase();
-        expect(managementKeys).to.have.lengthOf(1);
-        expect(managementKeys[0]).to.endsWith(expectedKey);
-      });
+    it('should emit created event', async () => {
+      expect(callback).to.be.calledWith(sinon.match(transaction));
+    });
+  });
 
-      it('has ENS name reserved', async () => {
-        expect(await provider.resolveName('alex.mylogin.eth')).to.eq(contract.address);
-      });
+  describe('Execute signed', async () => {
+    it('Error when not enough tokens', async () => {
+      const signature = calculateMessageSignature(wallet, {...msg, value: utils.parseEther('100')});
+      expect(identityService.executeSigned(contract.address, {...msg, signature})).to.be.eventually.rejected;
```

Comment:
`rejectedWith`
btw would be good to make sure proper http code is send if exception is thrown